File: C++/题目1512：用两个栈实现队列.cpp

#include <stdio.h>
#include <stack>   
#include <string.h>
#include <string>
using namespace std; 
int n,i; 
 
int main(){
    while(scanf("%d",&n) != EOF){
        stack<int> stack1;
        stack<int> stack2;
        for(i = 0; i < n; i++){
            char operate[5];
            scanf("%s",operate);
            if(strcmp(operate,"PUSH") == 0){
				int k;
                scanf("%d",&k);
                stack1.push(k);
            }else{
                if(!stack2.empty()){
					int top = stack2.top();	
					printf("%d\n",top);
					stack2.pop();
				}else{
					if(stack1.empty()){
						printf("-1\n");
					}else{
						while(!stack1.empty()){
							int top = stack1.top();
							stack1.pop();
							stack2.push(top);
						}
						printf("%d\n",stack2.top());
						stack2.pop();
					}
				}
            }
        }
         
    }
    return 0;
} 
 

/**************************************************************
	Problem: 1512
	User: wangzhenqing
	Language: C++
	Result: Accepted
	Time:70 ms
	Memory:1184 kb
****************************************************************/


                        

File: Java/题目1512：用两个栈实现队列.java

// 题目1512：用两个栈实现队列

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:55
 * @url:http://ac.jobdu.com/problem.php?pid=1512
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17243695
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.StreamTokenizer;
import java.util.Stack;

public class Main {
    /*
     * 1522
     */
public static void main(String[] args) throws Exception {
        StreamTokenizer st = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
        while (st.nextToken() != StreamTokenizer.TT_EOF) {
            int n = (int) st.nval;
            Stack<Integer> stack1 = new Stack<Integer>();
            Stack<Integer> stack2 = new Stack<Integer>();
            for (int i = 0; i < n; i++) {
            	st.nextToken();
                String a = st.sval;
                if (a.equals("POP")) {
                    if (stack2.isEmpty()) {
                        while(!stack1.isEmpty()){
                            stack2.push(stack1.pop());
                        }
                    }
                    if (stack2.isEmpty()) {
                        System.out.println(-1);
                    }else {
                        System.out.println(stack2.pop());
                    }
                }else if (a.contains("PUSH")) {
                	st.nextToken();
                    int tempNum = (int) st.nval;
                    stack1.push(tempNum);
                }
            }
        }
    }}
/**************************************************************
 Problem: 1512
 User: wangzhenqing
 Language: Java
 Result: Accepted
 Time:1370 ms
 Memory:42204 kb
 ****************************************************************/




